src/tqec/templates/scalable/rectangle.py

```diff
@@ -4,19 +4,36 @@
 
 
 class ScalableRectangle(ScalableTemplate):
-    def __init__(self, width: int, height: int) -> None:
+    def __init__(
+        self, 
+        width: int, 
+        height: int, 
+        scale_width: bool | None = None,
+    ) -> None:
+        """Rectangle with scalable width **or** height.
+
+        A scalable rectangle can only scale its width **or** height, but not both.
+
+        :param width: width of the rectangle.   
+        :param height: height of the rectangle.
+        :param scale_width: whether to scale the width or height. If None, the dimension
+            with the even value or the larger value will be scaled. If both dimensions
+            are even and equal, the width will be scaled by default.
+        """
+        assert (width % 2 == 0) or (height % 2 == 0), "At least one dimension must be even to be scalable!"
+        # Determine which dimension to scale
+        if scale_width is None:
+            scale_width = height % 2 != 0 or (width % 2 == 0 and width >= height)
         super().__init__(Rectangle(width, height))
+        self._scale_width = scale_width
```

Comment:
The type annotation is missing here. Most type checkers will be able to infer, but some might incorrectly infer `bool | None` instead of `bool`, so let's be explicit.
```suggestion
        self._scale_width: bool = scale_width
```